File: DesignPatterns/src/Java/003-Other-Patterns/004-MVC/MVC/src/org/csystem/app/entity/Person.java
package org.csystem.app.entity;

import java.io.Serializable;
import java.time.LocalDate;

public class Person implements Serializable {
    private int m_id;
    private String m_name;
    private LocalDate m_registerDate;

    public Person(int id, String name, LocalDate registerDate)
    {
        m_id = id;
        m_name = name;
        m_registerDate = registerDate;
    }

    public int getId()
    {
        return m_id;
    }

    public void setId(int id)
    {
        m_id = id;
    }

    public String getName()
    {
        return m_name;
    }

    public void setName(String name)
    {
        m_name = name;
    }

    public LocalDate getRegisterDate()
    {
        return m_registerDate;
    }

    public void setRegisterDate(LocalDate registerDate)
    {
        m_registerDate = registerDate;
    }

    @Override
    public String toString()
    {
        return String.format("[%d]%s-%s", m_id, m_name, m_registerDate);
    }
}

File: src/DesignPatterns/src/C++/002-GOFPatterns/002-StructuralPatterns/007-Bridge/PersonImpl/Person.cpp
#include "Person.h"
#include <string>

struct PersonImpl {
	std::string m_name;
	int m_no;	
};

using namespace std;

static std::string getName(PersonImpl *p)
{
	return p->m_name;
}

static int getNo(PersonImpl *p)
{
	return p->m_no;
}

Person::Person(const char *name, int no) : m_impl{ new PersonImpl }
{
	m_impl->m_name = name;
	m_impl->m_no = no;
}

Person::Person(const std::string &name, int no) : Person{ name.data(), no }
{}

Person::~Person()
{
	delete m_impl;
}

std::string Person::GetName() const 
{
	return getName(m_impl);
}
int Person::GetNo() const 
{ 
	return getNo(m_impl);
}